Expand configuration information in Data Prepper Getting started guide. (#3298)
opensearch-project/documentation-website: _data-prepper/getting-started.md

For Data Prepper previous to version 2.0, the `.jar` file expects the pipeline configuration file path to be followed by the server configuration file path. See the following configuration path example:

```
java -jar data-prepper-core-$VERSION.jar pipelines.yaml data-prepper-config.yaml
```

Optionally, you can add `"-Dlog4j.configurationFile=config/log4j2.properties"` to the command to pass a custom Log4j 2 configuration file. If you don't provide a properties file, Data Prepper defaults to the `log4j2.properties` file in the `shared-config` directory.

Starting with Data Prepper 2.0, you can launch Data Prepper by using the following `data-prepper` script that does not require any additional command line arguments:

```
bin/data-prepper
```

Configuration files are read from specific subdirectories in the application's home directory:
1. `pipelines/`: Used for pipeline configurations. Pipeline configurations can be written in one or more YAML files.
2. `config/data-prepper-config.yaml`: Used for the Data Prepper server configuration.

> You can supply your own pipeline configuration file path followed by the server configuration file path. However, support for this method will be dropped in a future release. See the following example:

Reviewer: ```suggestion
You can supply your own pipeline configuration file path followed by the server configuration file path. However, this method will not be supported in a future release. See the following example:
```

## 2. Configuring Data Prepper

> When you run a Data Prepper instance, it requires two configuration files to run, and also allows you to use an optional Log4j 2 configuration file. See [Configuring Log4j]({{site.url}}{{site.baseurl}}/data-prepper/managing-data-prepper/configuring-log4j/) for more information on configuring Log4j in Data Prepper. The following table explains the purpose of each configuration file.

Reviewer 2: ```suggestion
Two configuration files are required to run a Data Prepper instance. Optionally, you can configure a Log4j 2 configuration file. See [Configuring Log4j]({{site.url}}{{site.baseurl}}/data-prepper/managing-data-prepper/configuring-log4j/). The following list describes the purpose of each configuration file:
```

> Option | Required | Description

Reviewer 2: Since these are not parameters, we should make this a list

:--- | :--- | :--- |
`pipelines.yaml` | Yes | This file describes which data pipelines to run, including sources, processors, and sinks. |

> `data-prepper-config.yaml` | Yes | This file contains Data Prepper server settings so that you can interact with exposed Data Prepper server APIs. |

Reviewer 2: ```suggestion
`data-prepper-config.yaml` | Yes | This file contains Data Prepper server settings that allow you to interact with exposed Data Prepper server APIs. |
```

`log4j2-rolling.properties` | No | This file contains Log4j 2 configuration options and can be a JSON, YAML, XML, or a .properties file type. |

> For Data Prepper previous to version 2.0, the `.jar` file expects the pipeline configuration file path to be followed by the server configuration file path. See the following configuration path example.

Reviewer 2: ```suggestion
For Data Prepper versions below 2.0, the `.jar` file expects the pipeline configuration file path to be followed by the server configuration file path. See the following configuration path example:
```

Author: We shouldn't use "below" unless it's referring to a location or direction.

Reviewer 3: ~I think I remember Fanit using this language, too. Which tells me it's acceptable in the software business. This might be a call for Nate. For what it's worth, I would rewrite it ...~

~"For Data Prepper versions prior to 2.0, ..." Earlier than / later than could work.~

Scratch that.
Well, Microsoft style:
"Use or earlier or previous instead of or lower to refer to all versions of a product that precede a particular release if the statement is accurate for all preceding releases. For example, don't use Windows 10 or earlier unless the statement is accurate for Windows 1.0."

Apple:
"Earlier or later: Use earlier or later, rather than lower or higher or newer or older.
To use Handoff, you need iOS 8 or later."

AWS style is with the crickets on this one.

```
java -jar data-prepper-core-$VERSION.jar pipelines.yaml data-prepper-config.yaml
```

> Optionally, you can add `"-Dlog4j.configurationFile=config/log4j2.properties"` to the command if you want to pass a custom Log4j 2 configuration file to the command. If you don't provide a properties file, Data Prepper defaults to the `log4j2.properties` file in the `shared-config` directory.

Reviewer 2: ```suggestion
Optionally, you can add `"-Dlog4j.configurationFile=config/log4j2.properties"` to the command to pass a custom Log4j 2 configuration file. If you don't provide a properties file, Data Prepper defaults to the `log4j2.properties` file in the `shared-config` directory.
```

> Starting with Data Prepper versions 2.0 or later, you can launch Data Prepper by using the following `data-prepper` script that does not require you to include any additional command line arguments.

Reviewer 2: ```suggestion
Starting with Data Prepper 2.0, you can launch Data Prepper by using the following `data-prepper` script that does not require any additional command line arguments:
```

Reviewer 2: Only use a period before a table.

```
bin/data-prepper
```

Configuration files are read from specific subdirectories in the application's home directory:

> 1. `pipelines/`: Used for pipelines configurations. Pipelines configurations can be written in one or more yaml files.

Reviewer 3: I don't think "pipelines" plural is a proper noun here. The way I read this is that `pipelines/` indicates you can list more than one configuration files, but you can also have just one.
We refer to them singularly throughout our OpenSearch documentation as a single pipeline or multiple pipelines. So you could have one pipeline configurations or many pipeline configurations. In other words, I think "Used for pipeline configurations" is appropriate.

YAML

> 1. `pipelines/`: for pipelines configurations; pipelines configurations can be written in one and more yaml files

Reviewer 2: ```suggestion
1. `pipelines/`: Used for pipelines configurations. Pipelines configurations can be written in one or more yaml files.
```

> 2. `config/data-prepper-config.yaml`: for Data Prepper server configurations

Reviewer 2: ```suggestion
2. `config/data-prepper-config.yaml`: Used for the Data Prepper server configuration.
```

> You can continue to supply your own pipeline configuration file path followed by the server configuration file path, but the support for this method will be dropped in a future release. Example:

Reviewer 2: ```suggestion
You can supply your own pipeline configuration file path followed by the server configuration file path. However, support for this method will be dropped in a future release. See the following example:
```

```
bin/data-prepper pipelines.yaml data-prepper-config.yaml
```

> Additionally, the Log4j 2 configuration file is read from `config/log4j2.properties` in the application's home directory.

Reviewer 2: ```suggestion
The Log4j 2 configuration file is read from the `config/log4j2.properties` file located in the application's home directory.
```

## 3. Defining a pipeline

> Create a Data Prepper pipeline called `pipelines.yaml` and use the following configuration:

Reviewer 2: ```suggestion
Create a Data Prepper pipeline file named `pipelines.yaml` using the following configuration:
```
